### PR TITLE
feat: add custom errorHandler support

### PR DESCRIPTION
**Pull request by ngot**
**Summary**

- add the optional error handler config for custom error process.
- remove the Sinon since we already use jest.
- Correct the readme about the usage.
- fix some wrong test case.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -22,6 +22,8 @@ export interface Config {
    * default: ['/']
    */
   validatePaths: string[];
+
+  errorHandler?: (error: Error) => {};
```
Comment (supertong) [inline src/config.ts]: We don't need the extra `?` here. We have `Partial<Config>` already.

File: src/index.ts [hunk 2]
```diff
@@ -46,6 +46,9 @@ export function oas(cfg: Partial<Config>): koa.Middleware {
         body: (ctx.request as RequestWithBody).body,
       })
     } catch (err) {
+      if (config.errorHandler) {
+        config.errorHandler(err);
+      }
```
Comment (supertong) [inline src/index.ts]: If the `errorHandler` doesn't throw an Error, the code will keep going here with the `ctx.throw()` though.
I think we can have a `defaultErrorHandler` which is like
```
const defaultErrorHandler(err) {
  if (err instanceOf ChowError) {
      ctx.throw...
   } else { throw err}
}
```

Comment (ngot) [reply src/index.ts]: Make sense to me.

**Comment by supertong** [comment]
👍 Looks good to me, but could you need to fix the conflict :)

**Comment by ngot** [comment]
@supertong done

File: src/config.ts [hunk 3]
```diff
@@ -22,10 +25,25 @@ export interface Config {
...
+  /**
+   * Optional custom error handler
+   */
+  errorHandler: (error: Error, ctx: Context)=> void,
+}
+
+function defaultErrorHandler(err: Error, ctx: Context) {
+  if (err instanceof ChowError) {
+    const json = err.toJSON();
+    ctx.throw(400, 'Request validation error', { expose: true, ...json });
```
Comment (supertong) [inline src/config.ts]: Sorry, didn't notice the message only valid for Request, not Response. Hmm...

Comment (ngot) [reply src/config.ts]: Have updated this.

**Comment by supertong** [comment]
Thanks mate, looks good!

**Pull request merged**